cli/cmdlineparser.cpp

```diff
@@ -881,6 +881,7 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
                 }
             }
 
+            // TODO: deprecate "--template <template>"
```

Comment:
why would we deprecate `--template` what is the alternative?

Reply:
`--template=<template>`

As the added tests show it is prone to error and needs additional checking.

I think we should deprecate most of the options with whitespaces.

Reply:
ok sure.